errbot/__init__.py

```diff
@@ -397,4 +408,20 @@ def decorate(func):
 
     if len(args):
         return decorate(args[0], **kwargs)
-    return lambda func: decorate(func, **kwargs)
+    return lambda func: decorate(func)
+
+
+def botflow(*args, **kwargs):
+    """
+    Decorator for flow of commands.
+
+    TODO(gbin): example / docs
+    """
+    def decorate(func):
+        if not hasattr(func, '_err_flow'):  # don't override generated functions
+            func._err_flow = True
+        return func
+
+    if len(args):
+        return decorate(args[0], **kwargs)
```

Comment:
args[0] is the function you want to decorate, then we took the convention to only use kwargs for the optional arguments of the decorator